ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkTable/NetworkTable.tsx

```diff
@@ -64,6 +71,21 @@ const generateRows = (
     const numaNodes = getInterfaceNumaNodes(machine, nic);
     const vlan = vlans.find(({ id }) => id === nic.vlan_id);
     const fabric = vlan ? fabrics.find(({ id }) => id === vlan.fabric) : null;
+    const discoveredSubnetId =
+      nic?.discovered?.length && nic.discovered.length > 0
+        ? nic.discovered[0].subnet_id
+        : null;
+    const showSubnetLinks = fabric && !discoveredSubnetId;
+    const showSubnetDisplay = isAllNetworkingDisabled && discoveredSubnetId;
+    let subnetId: Subnet["id"] | null | undefined;
+    if (showSubnetLinks) {
+      // Look for a link to a subnet.
+      const subnetLink = nic?.links.find(({ subnet_id }) => !!subnet_id);
```

Comment:
Can a `subnet_id` be `0`?

Reply:
Good point!

Reply:
Actually, I think the subnet ids start at 1, but in MAAS `subnet_id` is created from other objects and not directly from the db so it might be safer to check for undefined/null anyway.